tvbwidgets/ui/ts_widget.py

```diff
@@ -0,0 +1,339 @@
+    def create_raw_from_ts(self):
+        # create Info object for Raw object
+        raw_info = mne.create_info(self.ch_names, sfreq=self.data.sample_rate)
+
+        valid_state_var = self.selected_state_var is not None
+        valid_mode = self.selected_mode is not None
+        if not (valid_state_var and valid_state_var):       # when plot is drawn for first time
+            data_for_raw = self.data.data[:, 0, :, 0]
+        else:
+            state_var = self.selected_state_var if valid_state_var else 0
+            mode = self.selected_mode if valid_mode else 0
+            data_for_raw = self.data.data[:, state_var, :, mode]
+
+        data_for_raw = np.swapaxes(data_for_raw, 0, 1)
+        raw = mne.io.RawArray(data_for_raw, raw_info)
+        self.raw = raw
+
+    # ========================================== BUTTONS ===============================================================
+    # buttons methods
+    def unselect_all(self, btn):
+        for cb_name in self.checkboxes:
+            self.checkboxes[cb_name].value = False
+
+    def select_all(self, btn):
+        for cb_name in self.checkboxes:
+            self.checkboxes[cb_name].value = True
+
+    # ===================================== INSTRUCTIONS DROPDOWN ======================================================
+    def create_instructions(self):
+        help_text = OrderedDict([
+            ('NAVIGATION', ' '),
+            ('→', f'Scroll ¼ right (scroll full window with Shift + →)'),
+            ('←', f'Scroll ¼ left (scroll full window with Shift + ←)'),
+            ('Home (fn + ← for Mac)', 'Show shorter time window'),
+            ('End (fn + → for Mac)', 'Show longer time window'),
+            ('↑', f'Scroll up (channels)'),
+            ('↓', f'Scroll down (channelss)'),
+            ('Page up (fn + ↑ for Mac)', 'Increase number of visible channels'),
+            ('Page down (fn + ↓ for Mac)', 'Decrease number of visible channels'),
+            ('SIGNAL TRANSFORMATIONS', ' '),
+            ('+ or =', 'Increase signal scaling'),
+            ('-', 'Decrease signal scaling'),
+            ('b', 'Toggle butterfly mode'),
+            ('d', 'Toggle DC removal'),
+            ('USER INTERFACE', ' '),
+            ('a', 'Toggle annotation mode'),
+            ('shift+j', 'Toggle all SSPs'),
+            ('p', 'Toggle draggable annotations'),
+            ('s', 'Toggle scalebars'),
+            ('z', 'Toggle scrollbars'),
+            ('esc', 'Close focused figure or dialog window'),
+            ('MOUSE INTERACTION', ' '),
+            (f'Left-click channel name', 'Mark/unmark bad channel'),
+            (f'Left-click channels data', 'Mark/unmark bad channel'),
+            ('Left-click-and-drag on plot', 'Add annotation (in annotation mode)'),
+            ('Left-click on plot background', 'Place vertical guide'),
+            ('Right-click on plot background', 'Clear vertical guide'),
```

Comment:
This relatively long text-based content would rather stay in a separate file. Maybe .ini or .properties ?